File: Python/Insert Interval.py
# Definition for an interval.
# class Interval(object):
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution(object):
    """
    :type intervals: List[Interval]
    :type newInterval: Interval
    :rtype: List[Interval]
    """
    def insert(self, intervals, newInterval):
        if intervals == None or newInterval == None:
            return intervals
            
        result = []
        pos    = 0
        
        for iv in intervals:
            if iv.end < newInterval.start:
                result.append(iv)
                pos += 1
            elif iv.start > newInterval.end:
                result.append(iv)
            else:
                newInterval.start = min(newInterval.start, iv.start)
                newInterval.end   = max(newInterval.end, iv.end)
                
        result.insert(pos, newInterval)
        
        return result
                

File: Java/Insert Interval.java
/**
 * Definition for an interval.
 * public class Interval {
 *     int start;
 *     int end;
 *     Interval() { start = 0; end = 0; }
 *     Interval(int s, int e) { start = s; end = e; }
 * }
 */
public class Solution {
    public List<Interval> insert(List<Interval> intervals, Interval newInterval) {
        if (newInterval == null || intervals == null)
            return intervals;
            
        List<Interval> result = new ArrayList<Interval>();
        int pos = 0;
        
        for (Interval iv : intervals) {
            if (iv.end < newInterval.start) {
                result.add(iv);
                pos++;
            } else if (iv.start > newInterval.end) {
                result.add(iv);
            } else {
                newInterval.start = Math.min(newInterval.start, iv.start);
                newInterval.end   = Math.max(newInterval.end, iv.end);
            }
        }
        
        result.add(pos, newInterval);
        
        return result;
    }
}